src/main/java/au/org/ala/biocache/dao/SearchDAOImpl.java

```diff
@@ -2937,25 +3215,40 @@ else if(fcount.getFacetField().getName().equals(COMMON_NAME_AND_LSID)){
         return new ArrayList<FacetResultDTO>(facetResults);
     }
 
+    @Override
     public Set<IndexFieldDTO> getIndexedFields() throws Exception{
         return getIndexedFields(false);
     }
 
     /**
      * Returns details about the fields in the index.
      */
-    public Set<IndexFieldDTO> getIndexedFields(boolean update) throws Exception{
-        if(indexFields == null || update){
-            indexFields = getIndexFieldDetails(null);
-            indexFieldMap = new HashMap<String, IndexFieldDTO>();
-            for(IndexFieldDTO field:indexFields){
-                indexFieldMap.put(field.getName(), field);
+    public Set<IndexFieldDTO> getIndexedFields(boolean update) throws Exception {
+        Set<IndexFieldDTO> result = indexFields;
+        if(result == null || update) {
+            synchronized(solrIndexVersionLock) {
+                result = indexFields;
+                if(result == null || update) {
+                    result = getIndexFieldDetails(null);
+                    Map<String, IndexFieldDTO> resultMap = new HashMap<String, IndexFieldDTO>();
+                    for(IndexFieldDTO field: result) {
+                        resultMap.put(field.getName(), field);
```

Comment:
This is not the only existing problem with ScatterplotController. There are also missing VALID_DATATYPES. 

A loop over indexFields instead of using indexFieldMap would work. This can be a new issue.